scripts/build-doc-list.js

```diff
@@ -7,20 +8,34 @@ const { slugify } = require('markdown-toc/lib/utils')
 const readingTime = require('reading-time')
 const { markdownToTxt } = require('markdown-to-txt')
 
-let specWeight = 100
-const result = []
+const yargs = require('yargs/yargs')
+const { hideBin } = require('yargs/helpers')
+const argv = yargs(hideBin(process.argv)).argv
+
 const basePath = 'pages'
-const postDirectories = [
+const defaultDirectoriesToProcess = [
   [`${basePath}/docs`, '/docs'],
   [`${basePath}/blog`, '/blog'],
   [`${basePath}/about`, '/about'],
-  [`${basePath}/jobs`, '/jobs'],
-]
-walkDirectories(postDirectories, result)
+  [`${basePath}/jobs`, '/jobs']
+];
+
+let specWeight = 100
+const result = []
+
+const output = argv.output;
+const groupCategoriesByFrontMatter = !!argv.groupByCategories;
+const directories = !argv.directory ? defaultDirectoriesToProcess : [[argv.directory]];
+
+walkDirectories(directories, result)
+
 if (process.env.NODE_ENV === 'production') {
   console.log(inspect(result, { depth: null, colors: true }))
 }
-writeFileSync(resolve(__dirname, '..', 'config', 'posts.json'), JSON.stringify(result, null, '  '))
+
+const groupedResults = groupCategoriesByFrontMatter ? checkAndGroupDocumentsByCategory(result) : result;
+
+writeFileSync(resolve(__dirname, '..', 'config', output), JSON.stringify(groupedResults, null, '  '))
```

Comment:
You should put that lines to separate function like `main()`:

```ts
function main() {
  walkDirectories(directories, result);
  
  if (process.env.NODE_ENV === 'production') {
    console.log(inspect(result, { depth: null, colors: true }))
  }
  
  const groupedResults = groupCategoriesByFrontMatter ? checkAndGroupDocumentsByCategory(result) : result;
  
  writeFileSync(resolve(__dirname, '..', 'config', output), JSON.stringify(groupedResults, null, '  '));
}
main();
```

Reply:
Cool, thanks , literally refactored stuff as you made this review, but I have added main to one of the missing files 👍